R/FDATask.R

```diff
@@ -0,0 +1,80 @@
+# @title Functional analysis task.
+#
+# @description
+# Converts a normal task to a FDA task by adding some extra information
+# (fd.features, fd.grids) to the task description and changing the S3
+# classes to FDATask. \dQuote{task.cl} and \dQuote{desc.cl} are the names for
+# the new class and task description respectively. It also does arg checking to
+# ensure consistency.
+#
+# @param task [\code{link{Task}}]\cr
+#   The normal mlr Task.
+# @param type [\code{character}]\cr
+#   The task type, will be set as a field to both task and task description. Either
+#   \code{fdaregr} or \code{fdaclassif}.
+# @template arg_fdatask_pars
+# @param task.cl [\code{character}]\cr
+#   The sub class name to convert to, eg. \code{FDAClassifTask} or \code{FDARegrTask}.
+# @param desc.cl [\code{character}]\cr
+#   The sub class description name to convert to, eg. \code{FDAClassifTaskDesc}
+#   or \code{FDARegrTaskDesc}.
+# @return An object of class \code{FDATask}.
+# @aliases FDATask
+convertTaskToFDATask = function(task, type, fd.features, fd.grids, task.cl, desc.cl) {
+  fnames = getTaskFeatureNames(task)
+  target = getTaskTargetNames(task)
+  # type could be fdaregr or fdaclassif
+  task$type = type
+  task$task.desc$type = type
+  assertList(fd.features, null.ok = TRUE, types = c("character", "integer"),
+    any.missing = FALSE, min.len = 1L, names = "unique")
+  assertList(fd.grids, null.ok = TRUE, types = "numeric",
+    any.missing = FALSE, min.len = 1L, names = "unique")
+  # if the user doesn't specify the fd.features, we assume all features belong to one group or one covariate
+  if (is.null(fd.features)) {
+    fd.features = list(fd1 = fnames)
+  }
+  if (is.null(fd.grids)) {
+    fd.grids = setNames(lapply(X = names(fd.features), FUN = function(name) {
+        as.numeric(1:length(fd.features[[name]]))
+      }), names(fd.features))
+  }
+  # check if target column is not used as fct.covariate
+  assert(target %nin% colnames(getTaskData(task)[, unlist(fd.features)]))
+  # check if length of each fct.covariate is equivalent to the length of its grid
+  lapply(names(fd.features), function(g) {
+    assert(length(fd.grids[[g]]) == length(fd.features[[g]]))
+  })
+  # we don't support mixed integer and character specification for fd.features
+  assertLogical(unique(vlapply(fd.features, is.integer)), len = 1)
+  assertNames(names(fd.grids), permutation.of = names(fd.features))
+  # two functional covariate can not occupy the same variable
+  assert(length(unlist(fd.features)) == length(unique(unlist(fd.features))))
+  cns = colnames(getTaskData(task))
```

Comment:
`cns = c(getTaskTargetNames(task), getTaskFeatureNames(task))` can't be used as I said in the previous comments which is already flushed by the commit, the problem is you can't decide the order in the right manner,  `cns = c(getTaskTargetNames(task), getTaskFeatureNames(task))`  and `cns = c( getTaskFeatureNames(task), getTaskTargetNames(task),)` makes a big difference, if we want to use this, we have to change our convention or forbid the user to use index